### DISCUSSION TITLE
Make project config more tolerant when adding fields without groups

### DISCUSSION THREAD

**Discussion by hiasl** [craftcms/cms#11522] [Ideas] [answered]
If you have an existing project and you want to copy/clone certain fields to another project, this is rather handy, just take the corresponding field and matrixBlockType yaml files from project-config, copy them to the new project and update the TS in project.yaml

But: if the corresponding field groups are not copied as well, it breaks the system, the field overview in CP's settings will not load anymore.

I hope it would be a rather small change to move those new fields to the common group if the related group does not exist and it would not beak Craft anymore.

**Comment by brandonkelly** [comment, marked as answer]
We can’t make changes to the project config as it’s being applied (like modifying the incoming field’s `fieldGroup` value).

However an invalid field group UID won’t stop the field from getting saved; it will just be missing its `groupId` afterward. So I’ve just made a change for the next release, to make Craft more defensive about the fields missing their `groupId`s (cb33c7448b96edc854c14845ceb0392e1a22fd0e). You won’t be able to add them to a field layout, but you will be able to access **Settings** → **Fields** without getting an error, and the field will be called out as “(Ungrouped)”:

<img src="https://user-images.githubusercontent.com/47792/176707327-d08140f7-c0e8-4eab-8432-dab79a1f193b.png" width="792" height="185" alt="A screenshot of the custom field index, with an ungrouped field highlighted">

From there you can edit the field and assign it to a group.

**Reply by hiasl** [reply]
This is perfect! Thank you.

**Reply by brandonkelly** [reply]
Craft 4.1.1 is out now with that change.
